### PR TITLE
Auth app

### PR DESCRIPTION
**Pull request by DrOncogene**
Restructured the backend to use application factory pattern
Added session-based authentication with flask_login
Added role-based access control
Modified tests to fit changes to models

### CODE REVIEW

File: web_backend/app/roles.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    @staticmethod
+    def allow(roles, methods=None):
+        """allows certain roles"""
+        def allow_decorator(view_func):
+            @wraps(view_func)
+            def wrapper(*args, **kwargs):
+                method_valid = request.method in methods if methods else True
+                role_valid = current_user and current_user.role in roles
+                if not (method_valid and role_valid):
+                    abort(401, description="Not priviledged")
```
Comment (Joseun) [inline web_backend/app/roles.py]: You could make this restricted access, instead of not priviledged and I think the status code is 403

**Review by Joseun — COMMENTED** [review]
We might need to discuss this via a video call. I think I understand some parts, but just to get a full idea. But you are doing great work.

File: web_backend/app/roles.py (hunk 1 above)
Comment (DrOncogene) [reply web_backend/app/roles.py]: Great. 403 is a better response. I'll add that

Comment (DrOncogene) [reply web_backend/app/roles.py]: Done

**Pull request merged**